setup.py

```diff
@@ -6,7 +6,7 @@
 from Cython.Build import cythonize
 
 
-if sys.version_info < (3, 12):
+if sys.platform == 'win32':
```

Comment:
We need < 3.12 on Windows as well because the numpy.distutils module is removed in Python 3.12. I guess it could be:
```
if sys.platform == 'win32' and sys.version_info < (3, 12)
```

Reply:
but does the case `sys.platform == 'win32'` and `sys.version_info >= (3, 12)` work at all?

Reply:
Yes, it works. Every CI job normally passes but the change here has caused the Windows job to fail when building wheels for Python 3.12. These same CI jobs are what build the wheels that go to PyPI which has Windows wheels for Python 3.12:
https://pypi.org/project/python-flint/#files

Some sort of change in CPython itself means that mingw+setuptools works for Python 3.12 but not for Python < 3.12. I don't follow all the details but the clearest explanation I found is here:
https://github.com/cython/cython/issues/4470#issuecomment-1361506709